app/validators/base_validator.rb

```diff
@@ -1,5 +1,7 @@
 class BaseValidator < ActiveModel::Validator
-  CASE_NUMBER_PATTERN ||= /^[BASTU](199|20\d)\d{5}$/i.freeze
+  CN_PATTERN ||= /^[BASTU](199|20\d)\d{5}$/i.freeze
+  URN_PATTERN ||= /^[A-Za-z0-9]{1,20}$/i.freeze
```

Comment:
may as well use full names for clarity

 CASE_URN_PATTERN ||=  /^[A-Za-z0-9]{1,20}$/i.freeze

Reply:
👍